### PR TITLE
Use internal machine pool for src build

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: eng/azure-pipelines.yml [hunk 1]
```diff
@@ -90,7 +90,10 @@ stages:
       - job: SourceBuild_Managed
         displayName: Source-Build (Managed)
         pool:
-          vmImage: ubuntu-latest
+          ${{ if eq(variables.officialBuild, 'false' )}}:
+            vmImage: ubuntu-latest
+          ${{ if eq(variables.officialBuild, 'true' )}}:
+            name: NetCore1ESPool-Internal -equals Build.Ubuntu.1804.Amd64
```
Comment (sbomer) [inline eng/azure-pipelines.yml:96]: How does the `-equals` syntax work here? Is it supposed to have a `demands` section?

Comment (vitek-karas) [reply eng/azure-pipelines.yml:96]: Based on Matt's template I think this should have:
```yaml
demands: ImageOverride -equals Build.Ubuntu.1804.Amd64
```

Comment (agocke) [reply eng/azure-pipelines.yml:96]: Ah, you're right, I misread that

**Review by vitek-karas — APPROVED** [review]
Looks good, now the question is: "How do we test this" ?
But I think we should merge - let it mirror and then see how it goes.

**Comment by agocke** [comment]
I know of no other way

**Review by sbomer — APPROVED** [review]

**Pull request merged**